src/pip/_internal/req/req_uninstall.py

```diff
@@ -267,14 +267,18 @@ def _get_file_stash(self, path):
     def stash(self, path):
         # type: (str) -> str
         """Stashes the directory or file and returns its new location.
+        Handle symlinks as files to avoid modifying the symlink targets.
         """
-        if os.path.isdir(path):
+        if os.path.isdir(path) and not os.path.islink(path):
```

Comment:
Can we assign this condition to something like `is_dir` and use it here and below?